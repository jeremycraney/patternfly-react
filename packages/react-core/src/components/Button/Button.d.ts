import { HTMLProps } from 'react';

export enum ButtonVariant {
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary',
  danger = 'danger',
  link = 'link',
  action = 'action'
}

export enum ButtonType {
  button = 'button',
  submit = 'submit'
}

export interface ButtonProps extends HTMLProps<HTMLButtonElement> {
  children?: React.ReactNode;
  isActive?: boolean;
  isBlock?: boolean;
  isDisabled?: boolean;
  isFocus?: boolean;
  isHover?: boolean;
  variant?: ButtonVariant;
  type?: ButtonType;
}

declare const Button: React.SFC<ButtonProps>;

export default Button;
