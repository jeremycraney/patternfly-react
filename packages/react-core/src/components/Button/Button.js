import React from 'react';
import { css, getModifier } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Button/styles.css';

export const ButtonVariant = {
  primary: 'primary',
  secondary: 'secondary',
  tertiary: 'tertiary',
  danger: 'danger',
  link: 'link',
  action: 'action'
};

export const ButtonType = {
  button: 'button',
  submit: 'submit'
};

const propTypes = {
  /** content rendered inside the button */
  children: PropTypes.node,
  /** additional classes added to the button */
  className: PropTypes.string,
  /**  Adds active styling to button. */
  isActive: PropTypes.bool,
  /** Adds block styling to button */
  isBlock: PropTypes.bool,
  /** Disables the button and adds disabled styling */
  isDisabled: PropTypes.bool,
  /** Adds focus styling to the button */
  isFocus: PropTypes.bool,
  /** Adds hove styling to the button */
  isHover: PropTypes.bool,
  /* Aria label used for action buttons that only use icons */
  label: PropTypes.string,
  /** Sets button type */
  type: PropTypes.oneOf(Object.keys(ButtonType)),
  /* Adds button variant styles */
  variant: PropTypes.oneOf(Object.keys(ButtonVariant))
};

const defaultProps = {
  children: '',
  className: '',
  isActive: false,
  isBlock: false,
  isDisabled: false,
  isFocus: false,
  isHover: false,
  label: '',
  type: ButtonType.button,
  variant: ButtonVariant.primary
};

const Button = ({
  children,
  className,
  isActive,
  isBlock,
  isDisabled,
  isFocus,
  isHover,
  label,
  variant,
  ...props
}) => (
  <button
    {...props}
    disabled={isDisabled}
    aria-label={variant === ButtonVariant.action ? label : null}
    className={css(
      styles.button,
      getModifier(styles.modifiers, variant),
      isBlock && styles.modifiers.block,
      isDisabled && styles.modifiers.disabled,
      isActive && styles.modifiers.active,
      isFocus && styles.modifiers.focus,
      isHover && styles.modifiers.hover,
      className
    )}
  >
    {children}
  </button>
);

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
