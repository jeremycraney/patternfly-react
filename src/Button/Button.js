import { Button as BsButton } from 'react-bootstrap'
import React from 'react'
import PropTypes from 'prop-types'

const Button = props => <BsButton {...props} />

Button.propTypes = Object.assign(BsButton.propTypes, {
  bsStyle: PropTypes.oneOf(['danger', 'default', 'primary', 'link'])
})

export default Button
