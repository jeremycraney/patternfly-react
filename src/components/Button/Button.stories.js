import React from 'react';
import { storiesOf } from '@storybook/react';
import { defaultTemplate } from '../../../storybook/decorators/storyTemplates';
import { Button, ButtonGroup } from './index';
import { Grid, Row, Col } from '../../index';

const stories = storiesOf('Button', module);

const description = (
  <p>
    This component is based on React Bootstrap Button component. See{' '}
    <a href="https://react-bootstrap.github.io/components.html#buttons">
      React Bootstrap Docs
    </a>{' '}
    for complete Button component documentation.
  </p>
);

stories.addDecorator(
  defaultTemplate({
    title: 'Button',
    documentationLink:
      'http://www.patternfly.org/pattern-library/widgets/#buttons',
    description: description,
  }),
);

stories.addWithInfo('Button', '', () => (
  <div>
    <p>
      <Button>Default Button</Button>{' '}
      <Button bsStyle="primary">Primary Button</Button>{' '}
      <Button bsStyle="danger">Danger Button</Button>{' '}
      <Button bsStyle="link">Link Button</Button>
    </p>
    <p>
      <Button bsSize="large">Large Button</Button>{' '}
      <Button>Default Button</Button>{' '}
      <Button bsSize="small">Small Button</Button>{' '}
      <Button bsSize="xsmall">Extra Small Button</Button>
    </p>
  </div>
));

stories.addWithInfo('ButtonGroup', () => {
  return (
    <Grid>
      <Row style={{ marginBottom: '20px' }}>
        <Col xs={12} md={3}>
          <ButtonGroup>
            <Button>Left</Button>
            <Button>Middle</Button>
            <Button>Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup>
            <Button bsStyle="danger">Left</Button>
            <Button bsStyle="danger">Middle</Button>
            <Button bsStyle="danger">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup>
            <Button bsStyle="primary">Left</Button>
            <Button bsStyle="primary">Middle</Button>
            <Button bsStyle="primary">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup>
            <Button bsStyle="link">Left</Button>
            <Button bsStyle="link">Middle</Button>
            <Button bsStyle="link">Right</Button>
          </ButtonGroup>
        </Col>
      </Row>
      <Row style={{ marginBottom: '20px' }}>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="large">
            <Button>Left</Button>
            <Button>Middle</Button>
            <Button>Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="large">
            <Button bsStyle="danger">Left</Button>
            <Button bsStyle="danger">Middle</Button>
            <Button bsStyle="danger">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="large">
            <Button bsStyle="primary">Left</Button>
            <Button bsStyle="primary">Middle</Button>
            <Button bsStyle="primary">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="large">
            <Button bsStyle="link">Left</Button>
            <Button bsStyle="link">Middle</Button>
            <Button bsStyle="link">Right</Button>
          </ButtonGroup>
        </Col>
      </Row>
      <Row style={{ marginBottom: '20px' }}>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="xsmall">
            <Button>Left</Button>
            <Button>Middle</Button>
            <Button>Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="xsmall">
            <Button bsStyle="danger">Left</Button>
            <Button bsStyle="danger">Middle</Button>
            <Button bsStyle="danger">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="xsmall">
            <Button bsStyle="primary">Left</Button>
            <Button bsStyle="primary">Middle</Button>
            <Button bsStyle="primary">Right</Button>
          </ButtonGroup>
        </Col>
        <Col xs={12} md={3}>
          <ButtonGroup bsSize="xsmall">
            <Button bsStyle="link">Left</Button>
            <Button bsStyle="link">Middle</Button>
            <Button bsStyle="link">Right</Button>
          </ButtonGroup>
        </Col>
      </Row>
    </Grid>
  );
});
