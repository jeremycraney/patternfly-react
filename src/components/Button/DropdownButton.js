import React from 'react';
import PropTypes from 'prop-types';
import { DropdownButton as BsDropdownButton } from 'react-bootstrap';
import { BUTTON_BS_STYLES } from './constants';

const DropdownButton = props => <BsDropdownButton {...props} />;

DropdownButton.propTypes = {
  ...BsDropdownButton.propTypes,
  bsStyle: PropTypes.oneOf(BUTTON_BS_STYLES),
};

DropdownButton.BUTTON_BS_STYLES = BUTTON_BS_STYLES;

export default DropdownButton;
