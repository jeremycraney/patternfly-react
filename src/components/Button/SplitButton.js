import React from 'react';
import PropTypes from 'prop-types';
import { SplitButton as BsSplitButton } from 'react-bootstrap';
import { BUTTON_BS_STYLES } from './constants';

const SplitButton = props => <BsSplitButton {...props} />;

SplitButton.propTypes = {
  ...BsSplitButton.propTypes,
  bsStyle: PropTypes.oneOf(BUTTON_BS_STYLES),
};

SplitButton.BUTTON_BS_STYLES = BUTTON_BS_STYLES;

export default SplitButton;
