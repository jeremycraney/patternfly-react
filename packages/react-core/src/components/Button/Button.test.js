import Button, { ButtonVariant } from './Button';
import React from 'react';
import { shallow } from 'enzyme';

Object.values(ButtonVariant).forEach(variant => {
  test(`${variant} button`, () => {
    const view = shallow(<Button variant={variant}>{variant} Button</Button>);
    expect(view).toMatchSnapshot();
  });
});

test('it adds an aria-label to action buttons', () => {
  const label = 'aria-label test';
  const view = shallow(<Button variant={ButtonVariant.action} label={label} />);
  expect(view.find('button').props()['aria-label']).toBe(label);
});

test('isBlock', () => {
  const view = shallow(<Button isBlock>Block Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isDisabled', () => {
  const view = shallow(<Button isDisabled>Disabled Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isFocus', () => {
  const view = shallow(<Button isFocus>Focused Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isHover', () => {
  const view = shallow(<Button isHover>Hovered Button</Button>);
  expect(view).toMatchSnapshot();
});
