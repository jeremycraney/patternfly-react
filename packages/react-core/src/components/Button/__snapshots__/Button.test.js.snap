// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action button 1`] = `
.pf-c-button.pf-m-action {
  display: inline-block;
  padding: 0.5rem 0.5rem 0.5rem 0.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border: 0px;
  border-radius: 30em;
  box-shadow: none;
  color: #292e34;
}

<button
  aria-label=""
  className="pf-c-button pf-m-action"
  disabled={false}
  type="button"
>
  action
   Button
</button>
`;

exports[`danger button 1`] = `
.pf-c-button.pf-m-danger {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #a30000;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #a30000;
  color: #ffffff;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-danger"
  disabled={false}
  type="button"
>
  danger
   Button
</button>
`;

exports[`isBlock 1`] = `
.pf-c-button.pf-m-primary.pf-m-block {
  display: block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #00659c;
  color: #ffffff;
  width: 100%;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-primary pf-m-block"
  disabled={false}
  type="button"
>
  Block Button
</button>
`;

exports[`isDisabled 1`] = `
.pf-c-button.pf-m-primary.pf-m-disabled {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 1px undefined;
  color: #ffffff;
  pointer-events: none;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-primary pf-m-disabled"
  disabled={true}
  type="button"
>
  Disabled Button
</button>
`;

exports[`isFocus 1`] = `
.pf-c-button.pf-m-primary.pf-m-focus {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #00659c;
  color: #ffffff;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-primary pf-m-focus"
  disabled={false}
  type="button"
>
  Focused Button
</button>
`;

exports[`isHover 1`] = `
.pf-c-button.pf-m-primary.pf-m-hover {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #00659c;
  color: #ffffff;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-primary pf-m-hover"
  disabled={false}
  type="button"
>
  Hovered Button
</button>
`;

exports[`link button 1`] = `
.pf-c-button.pf-m-link {
  display: inline-block;
  padding: 0.25rem 0.25rem 0.25rem 0.25rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border: 0px;
  border-radius: 30em;
  box-shadow: none;
  color: #00486e;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-link"
  disabled={false}
  type="button"
>
  link
   Button
</button>
`;

exports[`primary button 1`] = `
.pf-c-button.pf-m-primary {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #00659c;
  color: #ffffff;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-primary"
  disabled={false}
  type="button"
>
  primary
   Button
</button>
`;

exports[`secondary button 1`] = `
.pf-c-button.pf-m-secondary {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #00659c;
  color: #00659c;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-secondary"
  disabled={false}
  type="button"
>
  secondary
   Button
</button>
`;

exports[`tertiary button 1`] = `
.pf-c-button.pf-m-tertiary {
  display: inline-block;
  padding: 0.25rem 1.5rem 0.25rem 1.5rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border: 0px;
  border-radius: 30em;
  box-shadow: inset 0 0 0 2px #72767b;
  color: #00659c;
}

<button
  aria-label={null}
  className="pf-c-button pf-m-tertiary"
  disabled={false}
  type="button"
>
  tertiary
   Button
</button>
`;
